barretenberg/cpp/src/barretenberg/common/try_catch_shim.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+#include "barretenberg/common/log.hpp"
+#include <string>
+
+// Tool to make header only libraries (i.e. CLI11 and msgpack, though it has a bundled copy)
+// not uses exeptions with minimally invaslive changes
+
+#ifdef BB_NO_EXCEPTIONS
+struct __AbortStream {
+    void operator<< [[noreturn]] (const auto& error)
+    {
```

Comment:
Pulled from the msgpack lib copy we maintain as we now need this not just for msgpack